src/select.js

```diff
@@ -195,7 +195,7 @@
     ctrl.$filter = $filter;
 
     ctrl.isEmpty = function() {
-      return angular.isUndefined(ctrl.selected) || ctrl.selected === null || ctrl.selected === '';
+      return angular.isUndefined(ctrl.selected) || ctrl.selected === null || ctrl.selected === '' || ctrl.selected === 0;
```

Comment:
Not sure if this will be ok since `0` could be a value, so it isn't actually empty. This function will make the placeholder appear even when the user select `0`as value